docs/dev/go-style-guide.md

```diff
@@ -117,3 +118,200 @@ sdkerrors.Wrapf(
+    channelVersion string,
+    packet channeltypes.Packet,
+    relayer sdk.AccAddress,
+) ibcexported.Acknowledgement {
+
+// good
+func (im IBCMiddleware) OnRecvPacket(ctx sdk.Context, channelVersion string, packet channeltypes.Packet, relayer sdk.AccAddress) ibcexported.Acknowledgement {
+```
+
+---
+**Always write conditional statements like this**
+
+```go
+if i < f() {
+    g()
+}
+```
+
+Never like this
+
+```go
+if i < f()  // wrong! The compiler will place a semicolone (;) here.
+{           // wrong!
+    g()
+}
+```
+
+---
+**Don't Use Get in function/Method names**
+
+```go
+// bad
+
+// GetChainBID returns the chain-id for chain B.
+func (tc TestConfig) GetChainBID() string {
+    if tc.ChainConfigs[1].ChainID != "" {
+        return tc.ChainConfigs[1].ChainID
+    }
+    return "chainB-1"
+}
+
+// good
+func (tc TestConfig) ChainID(i int) string {
+    if tc.ChainConfigs[i].ChainID != "" {
+        return tc.ChainConfigs[i].ChainID
+    }
+    return "chainB-1"
+}
+```
+
+---
+**Do not make confusing indentation for saving vertical spaces**
+
+```go
+// Bad
+cases := []struct {
+		name     string
+		malleate func()
+		expErr   error
+	}{
+		{"verification success", func() {}, nil},
+		{"verification success: delay period passed", func() {
+			delayTimePeriod = uint64(1 * time.Second.Nanoseconds())
+		}, nil},
+		{"delay time period has not passed", func() {
+			delayTimePeriod = uint64(1 * time.Hour.Nanoseconds())
+		}, errorsmod.Wrap(ibctm.ErrDelayPeriodNotPassed, "failed packet commitment verification for client (07-tendermint-0): cannot verify packet until time: 1577926940000000000, current time: 1577923345000000000")},
+		{"client status is not active - client is expired", func() {
+			clientState, ok := path.EndpointB.GetClientState().(*ibctm.ClientState)
+			suite.Require().True(ok)
+			clientState.FrozenHeight = clienttypes.NewHeight(0, 1)
+			path.EndpointB.SetClientState(clientState)
+		}, errorsmod.Wrap(clienttypes.ErrClientNotActive, "client (07-tendermint-0) status is Frozen")},
+	}
+```
+
+```go
+// Bad
+{
+	"nil underlying app", func() {
+		isNilApp = true
+	}, nil,
+},
+```
+
+```go
+// Good
+{
+    "nil underlying app", 
+    func() {
+		    isNilApp = true
+    }, 
+    nil,
+},
+```
+
+## Good Practices
+
+**Testing context**
+
+Go 1.24 added a (testing.TB).Context() method. In tests, prefer using (testing.TB).Context() over context.Background() to provide the initial context.Context used by the test. Helper functions, environment or test double setup, and other functions called from the test function body that require a context should have one explicitly passed. [Referance](https://google.github.io/styleguide/go/decisions#contexts)
+
+## Known Anti Patterns
+
+It's strongly recommended [not to create a custom context.](https://google.github.io/styleguide/go/decisions#custom-contexts) But the cosmos sdk has it's own context that is passed around functions. And at this point it's too much of a pain to fix that.
```

Comment:
```suggestion
It's strongly recommended [not to create a custom context.](https://google.github.io/styleguide/go/decisions#custom-contexts). The cosmos sdk has it's own context that is passed around, and we should not try to work against that pattern to avoid confusion.
```

Or something like that.

Also, feels like there should be more anti-patterns? Maybe there aren't that many?